src/inmanta/data/model.py

```diff
@@ -475,3 +478,16 @@ class Agent(BaseModel):
     process_name: Optional[str]
     unpause_on_resume: Optional[bool]
     status: const.AgentStatus
+
+
+class AgentProcess(BaseModel):
+    sid: uuid.UUID
+    hostname: str
+    environment: uuid.UUID
+    first_seen: Optional[datetime.datetime]
+    last_seen: Optional[datetime.datetime]
+    expired: Optional[datetime.datetime]
+
+
+class AgentProcessDetails(AgentProcess):
+    state: Optional["ReportReturn"]
```

Comment:
Shouldn't it be this?
```suggestion
    state: Optional[Dict[str, ReportReturn]]
```

Reply:
Indeed